code401challenges/src/main/java/code401challenges/ArrayShift.java

```diff
@@ -1,7 +1,23 @@
 package code401challenges;
 
+import java.util.Arrays;
+
 public class ArrayShift {
-  public static int[] arrayShift(int[] arrToShift, int intToAdd){
+  public static void main(String[] args){
+    int[] inputArr = new int[]{4,8,15,23,42};
+    System.out.println(Arrays.toString(insertShiftArray(inputArr, 16)));
+  }
 
+  public static int[] insertShiftArray(int[] arrToShift, int intToAdd){
+    int[] newArr = new int[arrToShift.length+1];
+    double midIndex = Math.floor(arrToShift.length/2);
+    for(int i = 0; i < midIndex; i++){
+      newArr[i] = arrToShift[i];
+    }
+    newArr[(int)midIndex] = intToAdd;
+    for(int j = (int)midIndex + 1; j <= arrToShift.length; j++){
```

Comment:
To make your code more clear, I recommend casting when you are assigning a value to a variable instead of waiting until you're using that variable in the code